aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+
+This document demonstrates working with Forms and the HTML elements commonly used on a Form. The HTML `Form <https://www.w3.org/TR/html401/interact/forms.html>`__ element provides the primary mechanism web apps use to post back data to the server. Most of this document describes :doc:`Tag Helpers <tag-helpers/intro>` and how they can help you productively create robust HTML forms. We recommend you read :doc:`tag-helpers/intro` before you read this document.
+
+In many cases, :doc:`HTML Helpers </mvc/views/html-helpers>` provide an alternative approach to a specific Tag Helper, but it's important to recognize that Tag Helpers do not replace HTML Helpers and there is not a Tag Helper for each HTML Helper. When an HTML Helper alternative exists, it is mentioned.
+
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+The Form Tag Helper
+---------------------
+  
+The `Form <https://www.w3.org/TR/html401/interact/forms.html>`__ Tag Helper:
+
+- Generates the HTML `<FORM> <https://www.w3.org/TR/html401/interact/forms.html>`__ ``action`` attribute value for a MVC controller action or named route
+- Generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery (when used with the ``[ValidateAntiForgeryToken]`` attribute in the HTTP Post action method)
+- Provides the ``asp-route-<Parameter Name>`` attribute, where ``<Parameter Name>`` is added to the route values. The  ``routeValues`` parameter to ``Html.BeginForm`` and ``Html.BeginRouteForm`` provide similar functionality.  
+- Has an HTML Helper alternative ``Html.BeginForm`` and ``Html.BeginRouteForm``
+
+Sample:
+
+.. literalinclude::   forms/sample/final/Views/Demo/RegisterFormOnly.cshtml
+  :language: HTML
+
+The Form Tag Helper above generates the following HTML:
+ 
+.. code-block:: HTML
+
+  <form method="post" action="/Demo/Register">
+    <!-- Input and Submit elements -->
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+  
+The MVC runtime generates the ``action`` attribute value from the Form Tag Helper attributes ``asp-controller`` and ``asp-action``. The Form Tag Helper also generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery (when used with the ``[ValidateAntiForgeryToken]`` attribute in the HTTP Post action method). Protecting a pure HTML Form from cross-site request forgery is very difficult, the Form Tag Helper provides this service for you.
+
+Many of the views in the  *Views/Account* folder (generated when you create a new web app with *Individual User Accounts*) contain the `asp-route-returnurl <http://docs.asp.net/en/latest/mvc/views/working-with-forms.html#the-form-tag-helper>`__ attribute: 
```

Comment:
- Transition from previous paragraph remains a bit awkward.
- Still need a link back to the `asp-route-<Parameter Name>` feature.